src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -182,10 +182,11 @@ protected function processValue($value, bool $isRoot = false)
             foreach ($reflectionMethod->getParameters() as $key => $parameter) {
                 $names[$key] = $parameter->name;
 
-                if (\array_key_exists($key, $arguments) && '' !== $arguments[$key]) {
-                    continue;
+                if (\array_key_exists($parameter->name, $arguments)) {
+                    $key = $parameter->name;
                 }
-                if (\array_key_exists($parameter->name, $arguments) && '' !== $arguments[$parameter->name]) {
+
+                if (\array_key_exists($key, $arguments) && !\in_array($arguments[$key], ['', null, []], true)) {
```

Comment:
I'm not sure this makes sense: the empty string is a special case needed when loading from XML, but the others can only be defined by explicit definition.
It's also suspicious to me to change the DI component for an issue with the serializer one.

Reply:
@nicolas-grekas Yeah, I was way off with this solution, see #57273.